lib/transports.js

```diff
@@ -1,14 +1,16 @@
 var events = require('events');
 var util = require('util');
+var assign = require('object-assign');
```

Comment:
In my opinion, the use of such small modules seems more clearly, it's node way.
You worry about extra dependency?